src/Sessions.h

```diff
@@ -72,9 +71,11 @@ class NetSessions {
 	void GetStats(SessionStats& s) const;
 
 	void Weird(const char* name, const Packet* pkt,
-	           const EncapsulationStack* encap = nullptr, const char* addl = "");
+	           const std::shared_ptr<EncapsulationStack>& encap = nullptr,
+	           const char* addl = "");
 	void Weird(const char* name, const IP_Hdr* ip,
```

Comment:
Do we still need this version now? Wondering if callers could just do `Weird(name, packet)` and the method would take `ip_hdr` and `encap` out of `packet` if available. 

And, I guess, similar for the previous Weird: does it still need the `encap` argument given that that's probably be part of packet already (or not?)?

Reply:
I debated that, except for that I ran into one usage of it in PktSrc where they explicitly passed a nullptr for the third argument, and that caused a crash because addr was null. I had concerns that someone else would be calling that function directly and doing something similar.

Reply:
That's not to say that `Reporter::Weird` can't be fixed to properly handle null `addl` values.